Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/SparkJobHttpHandler.java

```diff
@@ -48,9 +50,13 @@ public void handle(HttpExchange httpExchange) throws IOException {
         try {
             String path = requestDetail.getRequestPath();
             if (path.equalsIgnoreCase("/applications/") && requestDetail.getAppId().equalsIgnoreCase("0")) {
-                List<Application> applications = JobViewCacheManager.getSparkApplications(requestDetail.getCluster());
-                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
-                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+                try {
+                    List<Application> applications = SparkRestUtil.getSparkApplications(requestDetail.getCluster());
+                    Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
+                    JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+                } catch (HDIException e) {
+                    DefaultLoader.getUIHelper().logError("get applications list error", e);
```

Comment:
You can try to implement ILogger interface and use log() to use slf4j to manage the errors.